dash/development/base_component.py

```diff
@@ -265,6 +312,64 @@ def __len__(self):
         return length
 
 
+def js_to_cerberus_type(type_object):
+    converters = {
+        'None': lambda x: {},
+        'func': lambda x: {},
+        'custom': lambda x: {},
+        'node': lambda x: {'type': 'component'},
```

Comment:
what should we do about `children`? Should component authors say that it's 
```
oneOfType([
     PropTypes.string, PropTypes.number, PropTypes.node, PropTypes.None,
     PropTypes.oneOf([PropTypes.string, PropTypes.number, PropTypes.node, PropTypes.None])
])
```

or something more complex?

Reply:
I think it would make sense to leave it up to authors to give more fine grained control. For example if there is a component that should only take a single dash component as a child (not strings or numbers) a user could use `PropTypes.element`.

For the more complicated things / things we already have validators for (like `figure`, we could use use the function validators pattern [here](http://docs.python-cerberus.org/en/stable/validation-rules.html#validator)).

To denote these in react, the `instanceOf` prop type would be useful. For example, something like `PropTypes.instanceOf(PlotlyFigure)` could trigger the figure validator to be used.

Reply:
Yeah, that sounds right. Looking at the `propTypes` docs, it seems like `children` should just always be a `node` and our `node` validation should handle all of the things that Dash can render
```
  // Anything that can be rendered: numbers, strings, elements or an array
  // (or fragment) containing these types.
  optionalNode: PropTypes.node,
```